growth/too/models.py

```diff
@@ -1007,3 +1012,247 @@ class Candidate(db.Model):
+
+    ssmagnr = db.Column(
+        db.Float,
+        nullable=True,
+        comment='Nearest solar system object mag')
+
+    ssdistnr = db.Column(
+        db.Float,
+        nullable=True,
+        comment='Nearest solar system object distance')
+
+    CLU_sfr_ha = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU star formation rate Halpha')
+
+    CLU_mstar = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU stellar mass')
+
+    CLU_sfr_fuv = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU star formation rate fuv')
+
+    CLU_z = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU redshift')
+
+    CLU_name = db.Column(
+        db.String,
+        nullable=True,
+        comment='CLU name')
+
+    CLU_id = db.Column(
+        db.Integer,
+        nullable=True,
+        comment='CLU ID')
+
+    CLU_d_to_galaxy_arcsec = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU distance arcsec')
+
+    comment = db.Column(
+        db.String,
+        nullable=True,
+        comment='Comment')
+
+    lightcurve = db.relationship(lambda: Lightcurve)
+
+
+class Lightcurve(db.Model):
+    """Candidate light curve pulled from the GROWTH
+    Marshal"""
+
+    name = db.Column(
+        db.ForeignKey(Candidate.name),
+        primary_key=True,
+        nullable=False,
+        comment='Candidate name')
+
+    date_observation = db.Column(
+        db.ARRAY(db.DateTime),
+        nullable=True,
+        comment='Observation date')
+
+    fil = db.Column(
+        db.ARRAY(db.String),
+        nullable=True,
+        comment='Filter')
+
+    instrument = db.Column(
+        db.ARRAY(db.String),
+        nullable=True,
+        comment='Instruments')
+
+    limmag = db.Column(
+        db.ARRAY(db.Float),
+        nullable=True,
+        comment='Limiting magnitude')
+
+    mag = db.Column(
+        db.ARRAY(db.Float),
+        nullable=True,
+        comment='Mag PSF')
+
+    magerr = db.Column(
+        db.ARRAY(db.Float),
+        nullable=True,
+        comment='Mag uncertainty')
+
+    exptime = db.Column(
+        db.ARRAY(db.Float),
+        nullable=True,
+        comment='Exposure time')
+
+    programid = db.Column(
+        db.ARRAY(db.Integer),
+        nullable=True,
+        comment='Program ID number (1,2,3)')
+
+    first_detection_time_tmp = db.Column(
+        db.DateTime,
+        nullable=True,
+        comment='Time first detection')
+
+    @hybrid_property
+    def first_detection_time(self):
+        if self.date_observation is None:
+            return None
+        jd_array = time.Time(self.date_observation,
+                             format='datetime').jd
+        min_jd = np.min(jd_array)
+        return time.Time(min_jd, format='jd').datetime
+
+    @first_detection_time.expression
+    def first_detection_time(cls):
+        return func.min(cls.date_observation)
```

Comment:
Won't this be the minimum over the entire table? That's probably not what you want.

Reply:
The table includes only detections, pre-filtered in growthdb_cgi.py.  So this expression should be OK.

Reply:
Please take a look at Plan.num_observations on master and follow that example. All of these aggregated properties should be properites of Candidate, not Lightcurve. Each of them should be a hybrid_property.